lib/logstash/filters/fingerprint.rb

```diff
@@ -120,12 +135,10 @@ def filter(event)
       if @concatenate_sources || @concatenate_all_fields
         to_string = ""
         if @concatenate_all_fields
-          event.to_hash.sort.map do |k,v|
-            to_string << "|#{k}|#{v}"
-          end
+          to_string << serialize(event)
```

Comment:
This would be a breaking change for any pipeline configuration that makes uses of `concatenate_all_fields`, because the string changes from pipe-delimited kv to json-ish.